event-handlers/src/main/java/handlers/UpsertNviCandidateHandler.java

```diff
@@ -0,0 +1,59 @@
+package handlers;
+
+import static java.util.Objects.isNull;
+import static no.unit.nva.commons.json.JsonUtils.dtoObjectMapper;
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage;
+import java.util.Objects;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpsertNviCandidateHandler implements RequestHandler<SQSEvent, Void> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpsertNviCandidateHandler.class);
+
+    private static final String ERROR_MESSAGE_BODY_INVALID = "Message body invalid";
+
+    @Override
+    public Void handleRequest(SQSEvent input, Context context) {
+        input.getRecords()
+            .stream()
+            .map(SQSMessage::getBody)
+            .map(this::parseBody)
+            .filter(Objects::nonNull)
+            .map(this::validate)
+            .forEach(this::upsertNviCandidate);
+
+        return null;
+    }
+
+    private void upsertNviCandidate(UpsertRequest request) {
+        //TODO: implement
+    }
+
+    private UpsertRequest parseBody(String body) {
+        return attempt(() -> dtoObjectMapper.readValue(body, UpsertRequest.class))
+                   .orElse(failure -> {
+                       logInvalidMessageBody(body);
+                       return null;
+                   });
+    }
+
+    //TODO: Remove jacocoGenerated when "happy cases" are implemented
+    @JacocoGenerated
+    private UpsertRequest validate(UpsertRequest request) {
+        if (isNull(request.publicationBucketUri())) {
+            logInvalidMessageBody(request.toString());
+            return null;
```

Comment:
I guess sending null to upsertNviCandidate method will cause exception, maybe throw exception already here instead of sending null to next method.

Reply:
The reason for not throwing an exception here is that SQSEvent input contains _multiple_ records. Therefore, if the message body of one record is invalid, returning null will make sure the lambda processes the _next_ record. If we throw an exception here, the next records will not be processed.

Reply:
https://docs.aws.amazon.com/lambda/latest/dg/with-sqs-create-package.html#with-sqs-example-deployment-pkg-java

Reply:
Okey, this handler will recieve sqs event from evaluate handler and as I see evaluate handler sends event with single record. But you are probably right.